core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -1484,4 +1486,36 @@ public void refreshEmotes(List<UUID> emotes) {
             player.sendUpstreamPacket(emoteList);
         }
     }
+
+    /**
+     * Send the following fog IDs, as well as the cached ones, to the client.
+     *
+     * Fog IDs can be found here:
+     * https://wiki.bedrock.dev/documentation/fog-ids.html
+     *
+     * @param fogNameSpaces the fog ids to add
+     */
+    public void sendFog(String... fogNameSpaces) {
+        this.fogNameSpaces.addAll(Arrays.asList(fogNameSpaces));
```

Comment:
Is there a way we can avoid the unnecessary asList allocation? Even if we have to do a for loop here. Same with removeFog below.

Reply:
When using a `for` loop IntelliJ actually bugs me to use `asList()`. This code will be called so rarely that I don't think it's worth changing and suppressing the warning.

Reply:
@Camotoy I can use `List.of` to make it immutable as that doesn't trigger a warning.

Reply:
IMO this is a non-issue - this code is not called very often and a single object allocation is negligible for performance. I would be more concerned if this code was called hundreds of time every tick.